### PR TITLE
getting started AB-Experiment

### PR DESCRIPTION
**Pull request by ivywei0125**
What does this implement/fix? Explain your changes.
---------------------------------------------------

Does this close any currently open issues?
------------------------------------------

Any relevant logs, screenshots, error output, etc.?
-------------------------------------

Any other comments?
-------------------
…

Has this been tested?
---------------------------
- [ ] Tested

### CODE REVIEW

File: Utils/azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/experiment/ExperimentationClient.java [hunk 1]
```diff
@@ -37,11 +37,17 @@ public static void init() {
 
     @Nullable
     public static String getFeatureVariable(String featureFlagName) {
+        if (!isInited) {
```
Comment (ivywei0125) [inline Utils/azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/experiment/ExperimentationClient.java:40]: add try catch to init()

File: Utils/azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/experiment/ExperimentationClient.java [hunk 2]
```diff
@@ -37,11 +37,17 @@ public static void init() {
...
+            init();
+        }
         return experimentationService.getFeatureVariable(featureFlagName);
     }
 
     @Nullable
     public static String getAssignmentContext() {
+        if (!isInited) {
```
Comment (ivywei0125) [inline Utils/azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/experiment/ExperimentationClient.java:48]: remove isInited, check service instance status

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/AzureActionsListener.java [hunk 3]
```diff
@@ -139,6 +140,7 @@ public void appFrameCreated(@NotNull List<String> commandLineArgs) {
 
         AzureStoreManager.register(new DefaultMachineStore(PluginHelper.getTemplateFile("azure.json")),
                 IntellijStore.getInstance(), IntelliJSecureStore.getInstance());
+        ExperimentationClient.init();
```
Comment (ivywei0125) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/AzureActionsListener.java:143]: remove init(). add getService()

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-guidance/src/main/java/com/microsoft/azure/toolkit/ide/guidance/view/CoursesView.java [hunk 4]
```diff
@@ -47,10 +47,10 @@ public class CoursesView {
 
     public CoursesView(@Nonnull Project project) {
         this.project = project;
+        this.showNewUIFlag = Boolean.parseBoolean(ExperimentationClient.getFeatureVariable(ExperimentationClient.FeatureFlag.GETTING_STARTED_UI.getFlagName()));
```
Comment (ivywei0125) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-guidance/src/main/java/com/microsoft/azure/toolkit/ide/guidance/view/CoursesView.java:50]: move showNewUIFlag to init() local var

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-guidance/src/main/java/com/microsoft/azure/toolkit/ide/guidance/view/CoursesView.java [hunk 5]
```diff
@@ -47,10 +47,10 @@ public class CoursesView {
...
         $$$setupUI$$$();
         init();
         JBList a;
```
Comment (ivywei0125) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-guidance/src/main/java/com/microsoft/azure/toolkit/ide/guidance/view/CoursesView.java:53]: remove JBList a

**Review by wangmingliang-ms — APPROVED** [review]
remember to resolve comments in later PRs.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-guidance/src/main/java/com/microsoft/azure/toolkit/ide/guidance/view/CoursesView.java (hunk 5 above)
Comment (ivywei0125) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-guidance/src/main/java/com/microsoft/azure/toolkit/ide/guidance/view/CoursesView.java:53]: resolved

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-guidance/src/main/java/com/microsoft/azure/toolkit/ide/guidance/view/CoursesView.java (hunk 4 above)
Comment (ivywei0125) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-guidance/src/main/java/com/microsoft/azure/toolkit/ide/guidance/view/CoursesView.java:50]: resolved

File: Utils/azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/experiment/ExperimentationClient.java (hunk 2 above)
Comment (ivywei0125) [reply Utils/azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/experiment/ExperimentationClient.java:48]: resolved

File: Utils/azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/experiment/ExperimentationClient.java (hunk 1 above)
Comment (ivywei0125) [reply Utils/azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/experiment/ExperimentationClient.java:40]: resolved

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/AzureActionsListener.java (hunk 3 above)
Comment (ivywei0125) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/AzureActionsListener.java:143]: resolved

**Pull request merged**